plugins/diff/windows/windows.go

```diff
@@ -245,10 +248,10 @@ func (s windowsDiff) Compare(ctx context.Context, lower, upper []mount.Mount, op
 		return emptyDesc, err
 	}
 
-	if isCompressed {
+	if compressionType != compression.Uncompressed {
 		dgstr := digest.SHA256.Digester()
 		var compressed io.WriteCloser
-		compressed, err = compression.CompressStream(cw, compression.Gzip)
+		compressed, err = compression.CompressStream(cw, compressionType)
```

Comment:
Do we need to extend `Extension` for `unknown` here? it could be friendly to see unknow error here, instead of empty

Reply:
Which extension? Unknown isn't used here

Reply:
https://github.com/containerd/containerd/blob/df06f44ecef00254bd4a1f86ac7333af7b36fe72/pkg/archive/compression/compression.go#L249

https://github.com/containerd/containerd/blob/df06f44ecef00254bd4a1f86ac7333af7b36fe72/pkg/archive/compression/compression.go#L254

> Unknown isn't used here

Sorry. comment on wrong line.
I was thinking on this line 146 in [plugins/diff/walking/differ.go](https://github.com/containerd/containerd/pull/11062/files#diff-d46efa4b3a14df904cb86578c9527d09a1eabba7d31b01769eaa4f3da2fd38f5).

Yeah, Unknown is not used. Since we use switch in Extension function and call `Extension` for error message, we should consider to extend `Extension`.

Reply:
I see what you mean, added